pkg/api/testapi/testapi.go

```diff
@@ -216,8 +225,11 @@ func (g TestGroup) InternalTypes() map[string]reflect.Type {
 
 // Codec returns the codec for the API version to test against, as set by the
 // KUBE_TEST_API_TYPE env var.
-func (g TestGroup) Codec() runtime.Codec {
+func (g TestGroup) Codec(gv *unversioned.GroupVersion) runtime.Codec {
 	if serializer.Serializer == nil {
+		if g.IsEnabledVersion(gv) {
```

Comment:
Yeah the overall lack of error handling or panic (it's just tests) is something I've struggled while working on this.